code/modules/mob/dead/new_player/new_player.dm

```diff
@@ -38,80 +40,138 @@
 /mob/dead/new_player/prepare_huds()
 	return
 
-/mob/dead/new_player/Topic(href, href_list[])
+/mob/dead/new_player/ui_interact(mob/user, datum/tgui/ui)
```

Comment:
This should be its own datum, not just put on top of new_player

Reply:
Yea, I've no idea what to do about getting that to work as it's own datum.

Reply:
I'm gonna assume I just call `ui_act` on it and it'll figure out the rest.

Reply:
we have lots of existing patterns of global datums with a ui. you call ui_interact on it